bugbug/models/rcatype.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import re
+
+import numpy as np
+import xgboost
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup
+from bugbug.model import BugModel
+
+# For the moment, rca - XYZ is treated of bugtype XYZ,
+# so we don't need to store it in a dictionary.
+RCA_CATEGORIES = [
+    "requirementerror",
+    "poorarchitecture",
+    "designerror",
+    "codingerror",
+    "testingerror",
+    "externalsoftwareaffectingfirefox",
+    "performanceerror",
+    "standards",
+    "systemerror",
+    "localizationerror",
+    "memory",
+    "infrastructure/builderror",
+    "communicationissues",
+    "productdecision",
+    "undocumentedchange",
+    "cornercase",
+]
+
+RCA_SUBCATEGORIES = [
+    "codingerror-syntaxerror",
+    "codingerror-logicalerror",
+    "codingerror-semanticerror",
+    "codingerror-runtimeerror",
+    "codingerror-unhandledexceptions",
+    "codingerror-internalapiissue",
+    "codingerror-networkissue",
+    "codingerror-compatibilityissue",
+    "codingerror-other",
+]
```

Comment:
In RCA_SUBCATEGORIES you don't have things like `cornercase` or `undocumentedchange`, you should still handle them when subcategories are enabled.
When subcategories are enabled, we want to be able to differentiate between all possible subcategories. If a category doesn't have any subcategory, it should still be taken into consideration.

Reply:
I think this comment still applies. When `rca_subcategories_enabled` is True, you are only considering subcategories, we should instead consider both.